src/tests/electron/tests/automated-checks.view.test.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Application } from 'spectron';
+import { createApplication } from 'tests/electron/common/create-application';
+import { dismissTelemetryOptInDialog } from 'tests/electron/common/dismiss-telemetry-opt-in-dialog';
+import { AutomatedChecksViewSelectors } from 'tests/electron/common/element-identifiers/automated-checks-view-selectors';
+import { CommonSelectors } from 'tests/electron/common/element-identifiers/common-selectors';
+import { testResourceServerConfig } from 'tests/electron/setup/test-resource-server-config';
+import { DEFAULT_ELECTRON_TEST_TIMEOUT_MS } from 'tests/electron/setup/timeouts';
+import * as WebDriverIO from 'webdriverio';
+
+// spectron wraps calls to electron APIs as promises. Unfortunately, only electron typings are used,
+// so tslint thinks some of the methods do not return promises.
+// tslint:disable: await-promise
+
+describe('AutomatedChecksView', () => {
+    let app: Application;
+    let client: WebDriverIO.Client<void>;
+
+    const collapsibleRuleDetailsGroupSelector = '.collapsible-rule-details-group';
+
+    beforeEach(async () => {
+        app = await createApplication();
+        client = app.client;
+    });
+
+    beforeEach(async () => {
+        await dismissTelemetryOptInDialog(app);
+
+        await ensureAppIsInDeviceConnectionDialog();
+
+        // inputs the port, validate and start scanning
+        await client.click(CommonSelectors.portNumber);
```

Comment:
Would it be better to add an `AutomatedChecksViewPage` type controller now like we have for the web pages?

Reply:
This is also a great point. I'm still concerned about how "usable" is to use spectron/webdriveio. I don't feel it the tests are easy to write considering I spend some good time figuring out how to get the elements I needed to click or verify. If you look at the selectors, they are no simple selectors (basically, div + class follow by another class or `li`), I didn't find a way to select and element and then select childrens from said element, which from my point of view would be ideal. The main reason for this is: webdriveio's Client does not return an actual html element (which have the `querySelector`/`querySelectorAll` function) but returns a webdriveio element wrapper.

So, to summaries, I agree that we should follow the "page controller" pattern we already have on the browser extension but I'm still worried about what spectron/webdriveio allows to do for writing 2e2 tests